From c130e5e4e1f2903e2cada860b5c6efed5e177d9b Mon Sep 17 00:00:00 2001
From: Martin Hauke <mardnh@gmx.de>
Date: Mon, 11 Nov 2019 20:03:02 +0100
Subject: [PATCH] build: fixes

Find and use system talloc, do not rely on OSMO's bundled copy.
Rename libmtp to libosmo-mtp, as libmtp is already used by the
Multimedia Transfer Protocol library.
---
 configure.ac           |  1 +
 libosmo-mtp.pc.in      |  2 +-
 libosmo-sccp.pc.in     |  2 +-
 src/Makefile.am        | 20 +++++++++++++++-----
 tests/m2ua/Makefile.am |  2 +-
 tests/sccp/Makefile.am |  3 ++-
 tests/ss7/Makefile.am  |  2 --
 tests/xua/Makefile.am  |  6 ++----
 8 files changed, 23 insertions(+), 15 deletions(-)

diff --git a/configure.ac b/configure.ac
index 25ff832..769e286 100644
--- a/configure.ac
+++ b/configure.ac
@@ -29,6 +29,7 @@ if test "x$PKG_CONFIG_INSTALLED" = "xno"; then
 fi
 PKG_PROG_PKG_CONFIG([0.20])
 
+PKG_CHECK_MODULES([TALLOC], [talloc])
 PKG_CHECK_MODULES(LIBOSMOCORE, libosmocore >= 1.3.0)
 PKG_CHECK_MODULES(LIBOSMOVTY, libosmovty >= 1.3.0)
 PKG_CHECK_MODULES(LIBOSMOGSM, libosmogsm >= 1.3.0)
diff --git a/libosmo-mtp.pc.in b/libosmo-mtp.pc.in
index 675d0d3..5e99dd3 100644
--- a/libosmo-mtp.pc.in
+++ b/libosmo-mtp.pc.in
@@ -6,5 +6,5 @@ includedir=@includedir@
 Name: Osmo MTP Lib
 Description: Osmo MTP Lib
 Version: @VERSION@
-Libs: -L${libdir} -lmtp
+Libs: -L${libdir} -losmo-mtp
 Cflags: -I${includedir}/
diff --git a/libosmo-sccp.pc.in b/libosmo-sccp.pc.in
index eda8d49..9dd18c1 100644
--- a/libosmo-sccp.pc.in
+++ b/libosmo-sccp.pc.in
@@ -6,5 +6,5 @@ includedir=@includedir@
 Name: OpenBSC SCCP Lib
 Description: OpenBSC SCCP Lib
 Version: @VERSION@
-Libs: -L${libdir} -lsccp
+Libs: -L${libdir} -losmo-sccp
 Cflags: -I${includedir}/
diff --git a/src/Makefile.am b/src/Makefile.am
index bdb225b..fe82fe9 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -7,15 +7,21 @@ noinst_HEADERS = sccp_internal.h xua_asp_fsm.h xua_as_fsm.h xua_internal.h
 # Legacy static libs
 
 sccpdir = $(libdir)
-sccp_LIBRARIES = libsccp.a libmtp.a libxua.a
+sccp_LTLIBRARIES = libosmo-sccp.la libosmo-mtp.la libosmo-xua.la
 
-libsccp_a_SOURCES = sccp.c
-libmtp_a_SOURCES = mtp_pcap.c
-libxua_a_SOURCES = xua_msg.c
+libosmo_sccp_la_SOURCES = sccp.c
+libosmo_mtp_la_SOURCES = mtp_pcap.c
+libosmo_xua_la_SOURCES = xua_msg.c
 # ensure that the file for the static lib is built with different C
 # flags, working around automake complaining that xua_msg.o is built
 # both with libtool (below) and without (here)
-libxua_a_CPPFLAGS = $(AM_CPPFLAGS) -DDUMMY -UDUMMY
+libosmo_xua_la_CPPFLAGS = ${AM_CPPFLAGS} -DDUMMY -UDUMMY
+
+libosmo_sccp_la_LIBADD = $(TALLOC_LIBS) $(LIBOSMOCORE_LIBS)
+libosmo_sccp_la_LDFLAGS = -release ${PACKAGE_VERSION}
+libosmo_mtp_la_LDFLAGS = -release ${PACKAGE_VERSION}
+libosmo_xua_la_LIBADD = $(TALLOC_LIBS) $(LIBOSMOCORE_LIBS)
+libosmo_xua_la_LDFLAGS = -release ${PACKAGE_VERSION}
 
 
 # New shared lib
@@ -35,3 +41,7 @@ libosmo_sigtran_la_SOURCES = sccp_sap.c sua.c m3ua.c xua_msg.c sccp_helpers.c \
 libosmo_sigtran_la_LDFLAGS = -version-info $(LIBVERSION) -no-undefined -export-symbols-regex '^osmo_'
 libosmo_sigtran_la_LIBADD = $(LIBOSMOCORE_LIBS) $(LIBOSMOGSM_LIBS) $(LIBOSMOVTY_LIBS) \
 			    $(LIBOSMONETIF_LIBS) $(LIBSCTP_LIBS)
+
+noinst_LTLIBRARIES = libosmo-sigtran-internal.la
+libosmo_sigtran_internal_la_SOURCES = $(libosmo_sigtran_la_SOURCES)
+libosmo_sigtran_internal_la_LIBADD = $(libosmo_sigtran_la_LIBADD)
diff --git a/tests/m2ua/Makefile.am b/tests/m2ua/Makefile.am
index 33618ef..a04145f 100644
--- a/tests/m2ua/Makefile.am
+++ b/tests/m2ua/Makefile.am
@@ -5,4 +5,4 @@ EXTRA_DIST = m2ua_test.ok
 
 noinst_PROGRAMS = m2ua_test
 m2ua_test_SOURCES = m2ua_test.c
-m2ua_test_LDADD = $(top_builddir)/src/libxua.a $(LIBOSMOCORE_LIBS)
+m2ua_test_LDADD = $(top_builddir)/src/libosmo-xua.la $(LIBOSMOCORE_LIBS) ${TALLOC_LIBS}
diff --git a/tests/sccp/Makefile.am b/tests/sccp/Makefile.am
index 85dbe64..ad7ddf3 100644
--- a/tests/sccp/Makefile.am
+++ b/tests/sccp/Makefile.am
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = $(all_includes) -I$(top_srcdir)/include
+AM_CPPFLAGS = $(all_includes) -I$(top_srcdir)/include ${TALLOC_CFLAGS}
 AM_CFLAGS=-Wall -ggdb3 $(LIBOSMOCORE_CFLAGS)
 
 EXTRA_DIST = sccp_test.ok
@@ -9,5 +9,6 @@ sccp_test_SOURCES = sccp_test.c
 sccp_test_LDADD = \
 	$(LIBOSMOCORE_LIBS) \
 	$(top_builddir)/src/sccp.o \
+        ${TALLOC_LIBS} \
 	$(NULL)
 
diff --git a/tests/ss7/Makefile.am b/tests/ss7/Makefile.am
index 3b6cb2c..d20b0a7 100644
--- a/tests/ss7/Makefile.am
+++ b/tests/ss7/Makefile.am
@@ -1,7 +1,5 @@
 AM_CPPFLAGS = $(all_includes) -I$(top_srcdir)/include -Wall
 AM_CFLAGS=-Wall $(LIBOSMOCORE_CFLAGS) $(LIBOSMOVTY_CFLAGS)
-
-AM_LDFLAGS = -static
 LDADD = $(top_builddir)/src/libosmo-sigtran.la \
 	$(LIBOSMOCORE_LIBS) $(LIBOSMOVTY_LIBS) $(LIBOSMONETIF_LIBS) $(LIBSCTP_LIBS)
 
diff --git a/tests/xua/Makefile.am b/tests/xua/Makefile.am
index f56692b..04dbe96 100644
--- a/tests/xua/Makefile.am
+++ b/tests/xua/Makefile.am
@@ -1,8 +1,6 @@
-AM_CPPFLAGS = $(all_includes) -I$(top_srcdir)/include -Wall
+AM_CPPFLAGS = $(all_includes) -I$(top_srcdir)/include -Wall $(LIBOSMOCORE_CFLAGS) $(LIBOSMOVTY_CFLAGS) $(LIBOSMONETIF_CFLAGS) $(LIBSCTP_CFLAGS)
 AM_CFLAGS=-Wall $(LIBOSMOCORE_CFLAGS) $(LIBOSMOVTY_CFLAGS)
-
-AM_LDFLAGS = -static
-LDADD = $(top_builddir)/src/libosmo-sigtran.la \
+LDADD = $(top_builddir)/src/libosmo-xua.la $(top_builddir)/src/libosmo-sigtran-internal.la \
 	$(LIBOSMOCORE_LIBS) $(LIBOSMOVTY_LIBS) $(LIBOSMONETIF_LIBS) $(LIBSCTP_LIBS)
 
 EXTRA_DIST = xua_test.ok xua_test.err
-- 
2.16.4

